test/integration/src/test/java/io/pravega/test/integration/endtoendtest/EndToEndTruncationTest.java

```diff
@@ -152,11 +166,292 @@ public void testTruncation() throws Exception {
+        StreamManager streamManager = StreamManager.create(controllerURI);
+        streamManager.createScope(scope);
+        streamManager.createStream(scope, streamName, streamConfiguration);
+        @Cleanup
+        ClientFactory clientFactory = ClientFactory.withScope(scope, controllerURI);
+
+        // Write totalEvents to the Stream.
+        writeDummyEvents(clientFactory, streamName, totalEvents);
+
+        // Instantiate readers to consume from Stream.
+        @Cleanup
+        ReaderGroupManager groupManager = ReaderGroupManager.withScope(scope, controllerURI);
+        groupManager.createReaderGroup(readerGroup, ReaderGroupConfig.builder().stream(Stream.of(scope, streamName)).build());
+        final List<CompletableFuture<Integer>> futures = readDummyEvents(clientFactory, readerGroup, parallelism);
+
+        // Wait some time to let readers read and then execute deletion.
+        Exceptions.handleInterrupted(() -> Thread.sleep(500));
+        assertTrue(streamManager.sealStream(scope, streamName));
+        assertTrue(streamManager.deleteStream(scope, streamName));
+
+        // At the control plane, we expect a RetriesExhaustedException as readers try to get successor segments from a deleted stream.
+        assertThrows(RetriesExhaustedException.class, () -> Futures.allOf(futures).join());
+        assertTrue(!streamManager.deleteStream(scope, streamName));
+    }
+
+    // start region utils
+
+    private List<CompletableFuture<Integer>> readDummyEvents(ClientFactory client, String rGroup, int numReaders, int limit) {
+        List<EventStreamReader<String>> readers = new ArrayList<>();
+        for (int i = 0; i < numReaders; i++) {
+            readers.add(client.createReader(String.valueOf(i), rGroup, new JavaSerializer<>(), ReaderConfig.builder().build()));
+        }
+
+        return readers.stream().map(r -> CompletableFuture.supplyAsync(() -> readEvents(r, limit))).collect(toList());
+    }
+
+    private List<CompletableFuture<Integer>> readDummyEvents(ClientFactory clientFactory, String readerGroup, int numReaders) {
+        return readDummyEvents(clientFactory, readerGroup, numReaders, Integer.MAX_VALUE);
+    }
+
+    private <T> int readEvents(EventStreamReader<T> reader, int limit) throws TruncatedDataException, RetriesExhaustedException {
+        final int timeout = 1000;
+        final int interReadWait = 50;
+        EventRead<T> event;
+        int validEvents = 0;
+        try {
+            do {
+                event = reader.readNextEvent(timeout);
+                Exceptions.handleInterrupted(() -> Thread.sleep(interReadWait));
+                if (event.getEvent() != null) {
+                    validEvents++;
+                }
+            } while ((event.getEvent() != null || event.isCheckpoint()) && validEvents < limit);
+
+            reader.close();
+        } catch (ReinitializationRequiredException e) {
+            e.printStackTrace();
```

Comment:
Do not print the stack trace, log it using slf4j. We already have the lombok annotation for this class.

Reply:
Missed that one. Now `ReinitializationRequiredException` are propagated to the caller.